### DISCUSSION TITLE
Tauri instead of electron?

### DISCUSSION THREAD

**Discussion by rp1231** [xanderfrangos/twinkle-tray#427] [Ideas]
I have recently been learning how to create web apps using tauri and it's pretty great as it gives a small executable compared to electron as it uses the systems native webview.

I think nodejs can be used by utilising the sidecar feature.
I'm not sure of the technicalities of whether this is feasible, but just thought I'd suggest this anyways.

**Comment by xanderfrangos** [comment]
Thanks for the suggestion!

While this could be worth looking into in the future, I have concerns about different versions of WebView2 introducing bugs. Electron, as bloated as it is, guarantees that the browser version will be identical for everyone. I would also need to replace non-browser functionality provided by Electron (such as the system tray and context menus). If someone provides a proof of concept, I'd be happy to look into it more. But right now I'm not in a rush to move away from Electron.

**Reply by rp1231** [reply]
> While this could be worth looking into in the future, I have concerns about different versions of WebView2 introducing bugs. Electron, as bloated as it is, guarantees that the browser version will be identical for everyone.

Yep, That is one of the benefits of electron.
One workaround would be that you could check the version of webview installed and then require installing/updating the web view.
But yes, it's not as convenient as electron in that regard.

>I would also need to replace non-browser functionality provided by Electron (such as the system tray and context menus). If someone provides a proof of concept, I'd be happy to look into it more. But right now I'm not in a rush to move away from Electron.

Tauri comes with that functionality out of the box and is pretty easy to do.(The system tray and context menus)
I'm not really a dev but was able to do it.
And tauri has plugins so you can position the webview window above the system tray 
https://github.com/JonasKruckenberg/tauri-plugin-positioner

This page has a list of the plugins:
https://github.com/tauri-apps/awesome-tauri#:~:text=GitHub%20Actions%20preconfigured.-,Plugins,-tauri%2Dplugin%2Dauthenticator

**Comment by xanderfrangos** [comment]
Good to know. I've looked at a few options previously, and Tauri seems like one of the better ones for my purposes. Thanks for sharing. 👍

**Comment by ruihe774** [comment]
I have made a [Tauri reimplementation](https://github.com/ruihe774/brightness-tray). It is very primitive (no settings page, no WMI support). You can have a try if you are interested in it.
